src/main/java/revai/StreamingClient.java

```diff
@@ -0,0 +1,205 @@
+package revai;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.WebSocket;
+import okhttp3.WebSocketListener;
+import okio.ByteString;
+import org.apache.http.client.utils.URIBuilder;
+import revai.helpers.SDKHelper;
+
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class StreamingClient {
+
+  private String accessToken;
+  private OkHttpClient client;
+  private WebSocket webSocket;
+  private String scheme;
+  private String host;
+  private StreamContentType streamContentType;
+  private String metadata;
+  private String customVocabularyId;
+  private Boolean filterProfanity;
+
+  /**
+   * Constructs the streaming client that is used to establish a websocket connection with the
+   * Rev.ai server and stream audio data. The user access token can be generated on the website at
+   * https://www.rev.ai/access_token. To establish a successful connection a valid StreamContentType
+   * must be provided. More info on the expected can be found at
+   * https://www.rev.ai/docs/streaming#section/WebSocket-Endpoint/Content-Type
+   *
+   * @param accessToken
+   * @param streamContentType
+   * @see StreamContentType
+   */
+  public StreamingClient(String accessToken, StreamContentType streamContentType) {
+    this.accessToken = accessToken;
+    this.streamContentType = streamContentType;
+    this.client = setClient();
+  }
+
+  /**
+   * This method is used for internal testing of the websocket connection against a mock webserver
+   * and should not be used
+   *
+   * @param scheme
+   */
+  public void setScheme(String scheme) {
+    this.scheme = scheme;
+  }
+
+  public String getHost() {
+    return host;
+  }
+
+  /**
+   * This method is used for internal testing and should not be used
+   *
+   * @param host
+   */
+  public void setHost(String host) {
+    this.host = host;
+  }
+
+  public String getMetadata() {
+    return metadata;
+  }
+
+  /**
+   * Setting metadata is optional but helps to identify the stream.
+   *
+   * @param metadata String size restricted to <= 256 characters
+   */
+  public void setMetadata(String metadata) {
+    this.metadata = metadata;
+  }
+
+  public String getCustomVocabularyId() {
```

Comment:
two questions:

- why are these various properties (metadata, vocabulary id, etc) provided as setters, whereas the contentType is provided in the constructor?
- and taking it a step further, why set them on the client at all instead of providing them as parameters of `connect` method.

all of them i think are just to construct the URL when connecting the socket, so i dont think they need to be stored as properties of the client object. 

in part i guess it depends on how you expect to use the client. if you expect to reuse it for multiple connections, then it would make sense i think to pass all those configuration options to the connect method, since they might differ between connections.

This is how node SDK does it (https://github.com/revdotcom/revai-node-sdk/blob/develop/src/streaming-client.ts#L65), though there for some reason content type is still specified as constructor param, which I am not sure why that is (cc: @kbridbur)

Reply:
I think this is a valid point. My reasoning behind this was mostly influenced by java's inability to make methods with optional parameters. I can easily overload that method to handle optional parameters.

Reply:
yeah its annoying there is no default parameter values.